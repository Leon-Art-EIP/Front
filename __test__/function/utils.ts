import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
