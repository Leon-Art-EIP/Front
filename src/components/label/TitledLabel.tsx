import { cn } from "../../tools/cn";

interface ITitledLabelProps {
  title: string;
  text: string;
  underline?: boolean;
  capitalize?: boolean;
}

export default function TitledLabel(props: ITitledLabelProps): JSX.Element {
  return (
    <div className="flex flex-col gap-4 font-semibold text-lg text-tertiary">
      <div>{props.title}</div>
      <div
        className={cn("px-4 text-secondary-tertiary", props.underline && "underline", props.capitalize && "capitalize")}
      >
        {props.text}
      </div>
    </div>
  );
}
