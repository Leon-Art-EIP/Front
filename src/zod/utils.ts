export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
