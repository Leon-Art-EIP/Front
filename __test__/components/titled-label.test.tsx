import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import TitledLabel from "../../src/components/label/TitledLabel";

test("IconLabel snapshot", () => {
  testSnapshot(<TitledLabel text="test" title="test" />);
});

test("IconLabel underline snapshot", () => {
  testSnapshot(<TitledLabel text="test" title="test" underline />);
});

test("IconLabel capitalize snapshot", () => {
  testSnapshot(<TitledLabel text="test" title="test" capitalize />);
});
