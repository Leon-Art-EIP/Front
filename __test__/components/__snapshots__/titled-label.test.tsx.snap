// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconLabel capitalize snapshot 1`] = `
<div
  className="flex flex-col gap-4 font-semibold text-lg text-tertiary"
>
  <div>
    test
  </div>
  <div
    className="px-4 text-secondary-tertiary font-normal capitalize"
  >
    test
  </div>
</div>
`;

exports[`IconLabel snapshot 1`] = `
<div
  className="flex flex-col gap-4 font-semibold text-lg text-tertiary"
>
  <div>
    test
  </div>
  <div
    className="px-4 text-secondary-tertiary font-normal"
  >
    test
  </div>
</div>
`;

exports[`IconLabel underline snapshot 1`] = `
<div
  className="flex flex-col gap-4 font-semibold text-lg text-tertiary"
>
  <div>
    test
  </div>
  <div
    className="px-4 text-secondary-tertiary font-normal underline"
  >
    test
  </div>
</div>
`;
